psi4/src/psi4/dlpno/sparse.cc

```diff
@@ -0,0 +1,219 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2021 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "sparse.h"
+
+#include <tuple>
+#include <algorithm>
+
+namespace psi {
+
+/* Args: sorted lists l1 and l2
+ * Returns: sorted union of l1 and l2
+ */
+std::vector<int> merge_lists(const std::vector<int> &l1, const std::vector<int> &l2) {
```

Comment:
Would it make sense to use `std::set<int>` instead of `std::vector<int>`? Then this function would just be `std::set_union`, but I don't know if this would create problems elsewhere.

Reply:
Using `std::set<int>` would also work here, though you'd have to cast to/from `std::vector<int>` or else rework the rest of the sparse map code to use `std::set<int>`. Besides the handy `set_union` function, I don't see any benefits to switching